enterprise/views.py

```diff
@@ -452,22 +482,44 @@ def get(self, request):
 
         # Get enterprise_customer to start in case we need to render a custom 404 page
         # Then go through other business logic to determine (and potentially overwrite) the enterprise customer
-        enterprise_customer = get_enterprise_customer_or_404(enterprise_customer_uuid)
+        try:
+            enterprise_customer = get_enterprise_customer_or_404(enterprise_customer_uuid)
+        except Http404:
+            error_code = 'ENTGDS008'
+            log_message = (
+                '[Enterprise DSC Flow] Received incorrect value for enterprise customer in request. '
+                'CourseId: {course_id}, '
+                'ProgramId: {program_uuid}, '
+                'EnterpriseCustomer: {enterprise_customer_uuid}, '
+                'User: {user_id}, '
+                'ErrorCode: {error_code} '.format(
+                    course_id=course_id,
+                    program_uuid=program_uuid,
+                    enterprise_customer_uuid=enterprise_customer_uuid,
+                    user_id=request.user.id,
+                    error_code=error_code,
+                )
+            )
+            LOGGER.error(log_message)
+            raise
+
         context_data = get_global_context(request, enterprise_customer)
 
         if not self.preview_mode:
             if not self.course_or_program_exist(course_id, program_uuid):
                 error_code = 'ENTGDS000'
                 log_message = (
-                    'Neither the course with course_id: {course_id} '
-                    'or program with {program_uuid} exist for '
-                    'enterprise customer {enterprise_customer_uuid}'
-                    'Error code {error_code} presented to user {userid}'.format(
+                    '[Enterprise DSC Flow] Neither the course or program exist for enterprise customer. '
+                    'CourseId: {course_id}, '
+                    'ProgramId: {program_uuid}, '
+                    'EnterpriseCustomer: {enterprise_customer_uuid}, '
+                    'User: {user_id}, '
+                    'ErrorCode: {error_code} '.format(
                         course_id=course_id,
                         program_uuid=program_uuid,
-                        error_code=error_code,
-                        userid=request.user.id,
                         enterprise_customer_uuid=enterprise_customer_uuid,
+                        user_id=request.user.id,
```

Comment:
In above log messages we were using username for `User:` here the id, we should be consistent